cpp/oneapi/dal/test/macro.hpp

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+// Disable clang-format as it dramatically
+// affects redability of macro definitions
+// clang-format off
+
+#define _TS_STRINGIFY(x) #x
+#define _TS_EXPAND(...)  __VA_ARGS__
```

Comment:
Usually all variables/functions/macros/etc that starts with underscore refers to internal implementation. Why there is no "public" macros in this file?

Reply:
I don't want this macros to be used explicitly in tests implementation. Their use should be limited by other more problem-specific macros, like `ALGO_TEST_CASE` (see `dal/test/common.hpp`) required to organize Catch2 test cases enumeration.

Reply:
Thanx for explanation!